# Proyek Analisis Data: [Input Nama Dataset]
- **Nama:** [Mayang Gumelar]
- **Email:** [gumelarmayang@gmail.com]
- **ID Dicoding:** [mayanggumelarr]

## Menentukan Pertanyaan Bisnis

1. Trend peminjaman setiap bulan baik dari casual maupun register?
2. Perbandingan peminjaman di hari biasa dan hari weekend?
3. Trend musim untuk peminjaman sepeda terbanyak dan tersedikit?
4. Di jam2 berapa peminjaman paling ramai dan paling sepi?

## Import Semua Packages/Library yang Digunakan

In [1]:
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
import seaborn as sns
import streamlit as st

## Data Wrangling

### Gathering Data

In [2]:
# data perhari
day_data = pd.read_csv("dataset/day.csv")
day_data.head()

instant    dteday  season  yr  mnth  holiday  weekday  workingday  \
0        1  1/1/2011       1   0     1        0        6           0   
1        2  1/2/2011       1   0     1        0        0           0   
2        3  1/3/2011       1   0     1        0        1           1   
3        4  1/4/2011       1   0     1        0        2           1   
4        5  1/5/2011       1   0     1        0        3           1   

   weathersit      temp     atemp       hum  windspeed  casual  registered  \
0           2  0.344167  0.363625  0.805833   0.160446     331         654   
1           2  0.363478  0.353739  0.696087   0.248539     131         670   
2           1  0.196364  0.189405  0.437273   0.248309     120        1229   
3           1  0.200000  0.212122  0.590435   0.160296     108        1454   
4           1  0.226957  0.229270  0.436957   0.186900      82        1518   

    cnt  
0   985  
1   801  
2  1349  
3  1562  
4  1600

In [3]:
# data perjam
hour_data = pd.read_csv("dataset/hour.csv")
hour_data.head()

instant    dteday  season  yr  mnth  hr  holiday  weekday  workingday  \
0        1  1/1/2011       1   0     1   0        0        6           0   
1        2  1/1/2011       1   0     1   1        0        6           0   
2        3  1/1/2011       1   0     1   2        0        6           0   
3        4  1/1/2011       1   0     1   3        0        6           0   
4        5  1/1/2011       1   0     1   4        0        6           0   

   weathersit  temp   atemp   hum  windspeed  casual  registered  cnt  
0           1  0.24  0.2879  0.81        0.0       3          13   16  
1           1  0.22  0.2727  0.80        0.0       8          32   40  
2           1  0.22  0.2727  0.80        0.0       5          27   32  
3           1  0.24  0.2879  0.75        0.0       3          10   13  
4           1  0.24  0.2879  0.75        0.0       0           1    1

**Insight:**
- data rental per-hari disimpan dalam bentuk dataframe dan ditampung dalam variabel day_data 
- data rental per-jam disimpan dalam bentuk dataframe dan ditampung dalam variabel hour_data
- terdapat perbedaan panjang data (banyak kolom) antara keduanya, ini karena pada hour_date terdapat kolom tambahan untuk 'jam'

### Assessing Data

In [4]:
# day_data info
day_data.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 731 entries, 0 to 730
Data columns (total 16 columns):
 #   Column      Non-Null Count  Dtype  
---  ------      --------------  -----  
 0   instant     731 non-null    int64  
 1   dteday      731 non-null    object 
 2   season      731 non-null    int64  
 3   yr          731 non-null    int64  
 4   mnth        731 non-null    int64  
 5   holiday     731 non-null    int64  
 6   weekday     731 non-null    int64  
 7   workingday  731 non-null    int64  
 8   weathersit  731 non-null    int64  
 9   temp        731 non-null    float64
 10  atemp       731 non-null    float64
 11  hum         731 non-null    float64
 12  windspeed   731 non-null    float64
 13  casual      731 non-null    int64  
 14  registered  731 non-null    int64  
 15  cnt         731 non-null    int64  
dtypes: float64(4), int64(11), object(1)
memory usage: 91.5+ KB


In [5]:
# statistik deskriptif day_data
print(f"Duplikasi data peminjaman perhari: {day_data.duplicated().sum()}")
day_data.describe(include="all")

Duplikasi data peminjaman perhari: 0


instant    dteday      season          yr        mnth     holiday  \
count   731.000000       731  731.000000  731.000000  731.000000  731.000000   
unique         NaN       731         NaN         NaN         NaN         NaN   
top            NaN  1/1/2011         NaN         NaN         NaN         NaN   
freq           NaN         1         NaN         NaN         NaN         NaN   
mean    366.000000       NaN    2.496580    0.500684    6.519836    0.028728   
std     211.165812       NaN    1.110807    0.500342    3.451913    0.167155   
min       1.000000       NaN    1.000000    0.000000    1.000000    0.000000   
25%     183.500000       NaN    2.000000    0.000000    4.000000    0.000000   
50%     366.000000       NaN    3.000000    1.000000    7.000000    0.000000   
75%     548.500000       NaN    3.000000    1.000000   10.000000    0.000000   
max     731.000000       NaN    4.000000    1.000000   12.000000    1.000000   

           weekday  workingday  weathersit        temp       atemp  \
count   731.000000  731.000000  731.000000  731.000000  731.000000   
unique         NaN         NaN         NaN         NaN         NaN   
top            NaN         NaN         NaN         NaN         NaN   
freq           NaN         NaN         NaN         NaN         NaN   
mean      2.997264    0.683995    1.395349    0.495385    0.474354   
std       2.004787    0.465233    0.544894    0.183051    0.162961   
min       0.000000    0.000000    1.000000    0.059130    0.079070   
25%       1.000000    0.000000    1.000000    0.337083    0.337842   
50%       3.000000    1.000000    1.000000    0.498333    0.486733   
75%       5.000000    1.000000    2.000000    0.655417    0.608602   
max       6.000000    1.000000    3.000000    0.861667    0.840896   

               hum   windspeed       casual   registered          cnt  
count   731.000000  731.000000   731.000000   731.000000   731.000000  
unique         NaN         NaN          NaN          NaN          NaN  
top            NaN         NaN          NaN          NaN          NaN  
freq           NaN         NaN          NaN          NaN          NaN  
mean      0.627894    0.190486   848.176471  3656.172367  4504.348837  
std       0.142429    0.077498   686.622488  1560.256377  1937.211452  
min       0.000000    0.022392     2.000000    20.000000    22.000000  
25%       0.520000    0.134950   315.500000  2497.000000  3152.000000  
50%       0.626667    0.180975   713.000000  3662.000000  4548.000000  
75%       0.730209    0.233214  1096.000000  4776.500000  5956.000000  
max       0.972500    0.507463  3410.000000  6946.000000  8714.000000

In [6]:
# hour_data info
hour_data.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 17379 entries, 0 to 17378
Data columns (total 17 columns):
 #   Column      Non-Null Count  Dtype  
---  ------      --------------  -----  
 0   instant     17379 non-null  int64  
 1   dteday      17379 non-null  object 
 2   season      17379 non-null  int64  
 3   yr          17379 non-null  int64  
 4   mnth        17379 non-null  int64  
 5   hr          17379 non-null  int64  
 6   holiday     17379 non-null  int64  
 7   weekday     17379 non-null  int64  
 8   workingday  17379 non-null  int64  
 9   weathersit  17379 non-null  int64  
 10  temp        17379 non-null  float64
 11  atemp       17379 non-null  float64
 12  hum         17379 non-null  float64
 13  windspeed   17379 non-null  float64
 14  casual      17379 non-null  int64  
 15  registered  17379 non-null  int64  
 16  cnt         17379 non-null  int64  
dtypes: float64(4), int64(12), object(1)
memory usage: 2.3+ MB


In [7]:
# statistik deskriptif hour_data
print(f"Duplikasi data peminjaman perhari: {hour_data.duplicated().sum()}")
hour_data.describe(include="all")

Duplikasi data peminjaman perhari: 0


instant      dteday        season            yr          mnth  \
count   17379.0000       17379  17379.000000  17379.000000  17379.000000   
unique         NaN         731           NaN           NaN           NaN   
top            NaN  12/31/2012           NaN           NaN           NaN   
freq           NaN          24           NaN           NaN           NaN   
mean     8690.0000         NaN      2.501640      0.502561      6.537775   
std      5017.0295         NaN      1.106918      0.500008      3.438776   
min         1.0000         NaN      1.000000      0.000000      1.000000   
25%      4345.5000         NaN      2.000000      0.000000      4.000000   
50%      8690.0000         NaN      3.000000      1.000000      7.000000   
75%     13034.5000         NaN      3.000000      1.000000     10.000000   
max     17379.0000         NaN      4.000000      1.000000     12.000000   

                  hr       holiday       weekday    workingday    weathersit  \
count   17379.000000  17379.000000  17379.000000  17379.000000  17379.000000   
unique           NaN           NaN           NaN           NaN           NaN   
top              NaN           NaN           NaN           NaN           NaN   
freq             NaN           NaN           NaN           NaN           NaN   
mean       11.546752      0.028770      3.003683      0.682721      1.425283   
std         6.914405      0.167165      2.005771      0.465431      0.639357   
min         0.000000      0.000000      0.000000      0.000000      1.000000   
25%         6.000000      0.000000      1.000000      0.000000      1.000000   
50%        12.000000      0.000000      3.000000      1.000000      1.000000   
75%        18.000000      0.000000      5.000000      1.000000      2.000000   
max        23.000000      1.000000      6.000000      1.000000      4.000000   

                temp         atemp           hum     windspeed        casual  \
count   17379.000000  17379.000000  17379.000000  17379.000000  17379.000000   
unique           NaN           NaN           NaN           NaN           NaN   
top              NaN           NaN           NaN           NaN           NaN   
freq             NaN           NaN           NaN           NaN           NaN   
mean        0.496987      0.475775      0.627229      0.190098     35.676218   
std         0.192556      0.171850      0.192930      0.122340     49.305030   
min         0.020000      0.000000      0.000000      0.000000      0.000000   
25%         0.340000      0.333300      0.480000      0.104500      4.000000   
50%         0.500000      0.484800      0.630000      0.194000     17.000000   
75%         0.660000      0.621200      0.780000      0.253700     48.000000   
max         1.000000      1.000000      1.000000      0.850700    367.000000   

          registered           cnt  
count   17379.000000  17379.000000  
unique           NaN           NaN  
top              NaN           NaN  
freq             NaN           NaN  
mean      153.786869    189.463088  
std       151.357286    181.387599  
min         0.000000      1.000000  
25%        34.000000     40.000000  
50%       115.000000    142.000000  
75%       220.000000    281.000000  
max       886.000000    977.000000

**Insight:**
- date_data terdiri dari 731 sampel, 16 feature, tidak ada missing values, tidak ada duplikasi
- hour_data terdiri dari 17.379 sampel, 17 feature, tidak ada missing values, tidak ada duplikasi
- feature dtday di kedua tabel masih bertipe data objek, seharusnya datetime
- semua nilai pada feature akurat dan tidak ada yang tidak masuk akal

### Cleaning Data

In [8]:
# cleaning data day_data
# konversi format dtday dari objek ke format datetime
day_data['dteday'] = pd.to_datetime(day_data['dteday'])

In [9]:
day_data.dteday.info()

<class 'pandas.core.series.Series'>
RangeIndex: 731 entries, 0 to 730
Series name: dteday
Non-Null Count  Dtype         
--------------  -----         
731 non-null    datetime64[ns]
dtypes: datetime64[ns](1)
memory usage: 5.8 KB


In [10]:
# cleaning data hour_data
# konversi format dtday dari objek ke format datetime
hour_data['dteday'] = pd.to_datetime(day_data['dteday'])

In [11]:
hour_data.dteday.info()

<class 'pandas.core.series.Series'>
RangeIndex: 17379 entries, 0 to 17378
Series name: dteday
Non-Null Count  Dtype         
--------------  -----         
731 non-null    datetime64[ns]
dtypes: datetime64[ns](1)
memory usage: 135.9 KB


In [12]:
# save data bersih
day_data.to_csv('day_clean.csv')
hour_data.to_csv('hour_clean.csv')

**Insight:**
- format object dteday pada data_day telah dirubah ke format datetime
- format object dteday pada hour_day telah dirubah juga ke format datetime
- tidak ada proses cleaning lain karena data terhitung bersih

## Exploratory Data Analysis (EDA)

### Explore ...

**Insight:**
- xxx
- xxx

## Visualization & Explanatory Analysis

### Pertanyaan 1:

### Pertanyaan 2:

**Insight:**
- xxx
- xxx

## Analisis Lanjutan (Opsional)

## Conclusion

- Conclution pertanyaan 1
- Conclution pertanyaan 2